### PR TITLE
MAINT: update Data Prepper documentation for 1.4

### PR DESCRIPTION
**Pull request by chenqi0805**
Signed-off-by: Chen <19492223+chenqi0805@users.noreply.github.com>

### Description
updated the following docs:

* data-prepper-reference.md: add record_type as new option in otel_trace_source; add otel_trace_raw as event record type counterpart of otel_trace_raw_prepper
* pipelines.md: add trace ingestion pipeline example for event record_type

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dlvenable — CHANGES_REQUESTED** [review]
Overall the changes look good. But, I'd like to see the documentation state very clearly to users that they should be migrating to the Event model. Data Prepper 2.0 will remove the old model. Users who start the migration in 1.4 will have an easier time updating to Data Prepper 2.0.

**Comment by chenqi0805** [comment]
> Overall the changes look good. But, I'd like to see the documentation state very clearly to users that they should be migrating to the Event model. Data Prepper 2.0 will remove the old model. Users who start the migration in 1.4 will have an easier time updating to Data Prepper 2.0.

Added a statement to indicate that in pipelines.md

**Review by dlvenable — APPROVED** [review]

File: _clients/data-prepper/data-prepper-reference.md [hunk 1]
```diff
@@ -37,22 +37,23 @@ Sources define where your data comes from.
...
+useAcmCertForSSL | No            | Boolean | Whether to enable TLS/SSL using certificate and private key from AWS Certificate Manager (ACM). Default is `false`.
+acmCertificateArn | Conditionally | String  | Represents the ACM certificate ARN. ACM certificate take preference over S3 or local file system certificate. Required if `useAcmCertForSSL` is set to `true`.
+awsRegion | Conditionally | String  | Represents the AWS region to use ACM or S3. Required if `useAcmCertForSSL` is set to `true` or `sslKeyCertChainFile` and `sslKeyFile` are AWS S3 paths.
+authentication | No            | Object  | An authentication configuration. By default, this runs an unauthenticated server. This uses pluggable authentication for HTTPS. To use basic authentication, define the `http_basic` plugin with a `username` and `password`. To provide customer authentication use or create a plugin which implements: [GrpcAuthenticationProvider](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/armeria-common/src/main/java/com/amazon/dataprepper/armeria/authentication/GrpcAuthenticationProvider.java).
```
Comment (keithhc2) [inline _clients/data-prepper/data-prepper-reference.md:55]: By "this runs an authenticated server", do you mean Data Prepper runs on an authenticated server? And enabling `authentication` will use pluggable authentication for HTTPS?

**Review by keithhc2 — APPROVED** [review]
Just some minor feedback.

**Comment by keithhc2** [comment]
@dlvenable Do you know if there will be any more docs updates coming with Data Prepper 1.4? Also, when is the release date for 1.4?

**Comment by dlvenable** [comment]
> @dlvenable Do you know if there will be any more docs updates coming with Data Prepper 1.4? Also, when is the release date for 1.4?

@keithhc2 , Yes, we will need to update the documentation with the new Metrics processor documentation. The release date is May 10.

File: _clients/data-prepper/pipelines.md [hunk 2]
```diff
@@ -75,6 +75,10 @@ This example uses weak security. We strongly recommend securing all plugins whic
 
 The following example demonstrates how to build a pipeline that supports the [Trace Analytics OpenSearch Dashboards plugin]({{site.url}}{{site.baseurl}}/observability-plugin/trace/ta-dashboards/). This pipeline takes data from the OpenTelemetry Collector and uses two other pipelines as sinks. These two separate pipelines index trace and the service map documents for the dashboard plugin.
 
+#### Classic
+
+This pipeline definition will be deprecated in 2.0. Users are recommended to use [Event record type](#event-record-type) pipeline definition.
```
[the remark this answers is not in the document]
Comment (dlvenable) [reply _clients/data-prepper/pipelines.md:80]: @keithhc2 , Yes, that will be the only option in 2.0.

File: _clients/data-prepper/data-prepper-reference.md (hunk 1 above)
Comment (dlvenable) [reply _clients/data-prepper/data-prepper-reference.md:55]: The authentication mechanisms is independent of HTTPS. The HTTPS is controlled by parameters such as `acmCertificateArn`.

Yes, the default pipeline will create an unauthenticated server, which means that any request is trusted.

File: _clients/data-prepper/pipelines.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (keithhc2) [reply _clients/data-prepper/pipelines.md:80]: Okay, I think that would be helpful to call out.

File: _clients/data-prepper/data-prepper-reference.md (hunk 1 above)
Comment (keithhc2) [reply _clients/data-prepper/data-prepper-reference.md:55]: Can we say that explicitly? Something like, "The default pipeline creates an unauthenticated server. To use basic authentication, define the ...".

**Comment by keithhc2** [comment]
> @keithhc2 , Yes, we will need to update the documentation with the new Metrics processor documentation. The release date is May 10.

Got it, I'll create the PR a few days before May 10 to merge in the 1.4 docs then. Thanks!

File: _clients/data-prepper/data-prepper-reference.md (hunk 1 above)
Comment (chenqi0805) [reply _clients/data-prepper/data-prepper-reference.md]: Changed wording to By default, this creates an unauthenticated server for the pipeline

**Pull request merged**